### PR TITLE
API Rewrite For 4.1

### PR DESCRIPTION
**Pull request by kabel**
Rewrite the API to support PHP 5.5 and get test coverage to 100%.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  fast_finish: true
+  include:
+    - php: 5.5
+    - php: 5.6
+      env:
+        - EXECUTE_TEST_COVERALLS=true
+    - php: 7
+    - php: hhvm
+  allow_failures:
+    - php: 7
```
Comment (mfairchild365) [inline .travis.yml]: I think 7 should be supported, is there a reason for allowing failures on it?

Comment (kabel) [inline .travis.yml]: Fair enough, I took it out.

File: src/Templates.php [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    }
+
+    /**
+     * Add a (java)script to the page.
+     *
+     * @param string $url  URL to the script
+     * @param string $type Type of script text/javascript
+     * @return self
+     */
+    public function addScript($url, $type = '')
```
Comment (mfairchild365) [inline src/Templates.php]: Just some thoughts: what about custom attributes, and/or adding a script to the end of the Body?

Comment (kabel) [inline src/Templates.php]: I'm not looking to _add_ to the API at this point. Just maintaining b/c. Adding to the end of the body is not currently supported by the framework.

**Comment by mfairchild365** [comment]
+10 for all the tests!

**Pull request merged**